Raptor integration - feature branch (#3121)
ezsystems/developer-documentation: docs/recommendations/raptor_integration/raptor_connector_guide.md

The SiteAccess-aware [Raptor connector](raptor_connector.md) provides a seamless integration between [[= product_name =]] and Raptor recommendation engine.

Its primary goal is to enable editors and managers to deliver personalized experiences across digital channels, which helps to increase conversion rates, drive sales, and improve user engagement.

By bringing content and recommendations together, the connector makes it easy to build and manage personalized experiences.

It provides a seamless integration layer that supports:

- event tracking of user interactions through the Tracking API
- delivery of tailored content through the Recommendations API
- flexible SiteAccess-aware configuration adapted to different sites and contexts

This approach simplifies integration while supporting personalization across different sites and markets.

## Availability

Raptor integration elements, such as tracking, twig functions, and public API, are available in all supported [[= product_name =]] editions starting from [[= latest_tag_5_0 =]] version.

Recommendation blocks provided in Page Builder, are available in [[= product_name_exp =]] edition.

## How does Raptor tracking work

> To start [tracking]((https://content.raptorservices.com/help-center/introduction-to-tracking-documentation)) user interactions, the tracking script needs to be added to the website’s layout.

Reviewer: Double parenthesis broke the link

> WARNING -  Doc file 'recommendations/raptor_integration/raptor_connector_guide.md' contains an unrecognized relative link '(https://content.raptorservices.com/help-center/introduction-to-tracking-documentation)', it was left as is.

```suggestion
To start [tracking](https://content.raptorservices.com/help-center/introduction-to-tracking-documentation) user interactions, the tracking script needs to be added to the website’s layout.
```

---
description:  Discover Raptor integration - an add-on focused on recommendations and tracking customer behaviors.
month_change: true
---

# Raptor integration guide

Discover [Raptor](https://www.raptorservices.com/) integration - an add-on that is focused on recommendations and tracking customer behaviors.
It includes Raptor connector with tracking scripts and events, used to track and analyze customer behaviors, and a set of Recommendation blocks.

## What is Raptor connector

> The SiteAccess-aware [Raptor connector](raptor_connector.md) provides a seamless integration between [[= product_name =]] and Raptor recommendation engine.

Reviewer 2: ```suggestion
The Raptor connector](raptor_connector.md) provides a seamless integration between [[= product_name =]] and Raptor recommendation engine.
```
In the DXP, almost everything is SiteAccess-aware

Its primary goal is to enable editors and managers to deliver personalized experiences across digital channels, which helps to increase conversion rates, drive sales, and improve user engagement.

By bringing content and recommendations together, the connector makes it easy to build and manage personalized experiences.

It provides a seamless integration layer that supports:

- event tracking of user interactions through the Tracking API

> - delivery of tailored content through the Recommendations API

Reviewer 2: I'd rewrite this to mention products somehow as well, as in the DXP content has a very specific meaning

- flexible SiteAccess-aware configuration adapted to different sites and contexts

This approach simplifies integration while supporting personalization across different sites and markets.

## Availability

> Raptor integration elements, such as tracking, twig functions, and public API, are available in all supported [[= product_name =]] editions starting from [[= latest_tag_5_0 =]] version.

Reviewer 2: ```suggestion
Raptor integration elements, such as tracking, Twig functions, and public API, are available in all supported [[= product_name =]] editions starting from v5.0.7 version.
```
Otherwise this value will keep increasing with each release

> Recommendation blocks provided in Page Builder, are available in [[= product_name_exp =]] edition.

Reviewer 2: ```suggestion
Recommendation blocks provided in Page Builder, are available in [[= product_name_exp =]] and [[= product_name_com =]] editions.
```

## How does Raptor tracking work

To start [tracking](https://content.raptorservices.com/help-center/introduction-to-tracking-documentation) user interactions, the tracking script needs to be added to the website’s layout.
Tracking can be set up either on the client-side or server-side, depending on how you want to capture and process the events.

The tracking works differently depending on the mode you choose.

> In server-side mode, it uses HTML placeholders and comments, handling all tracking without loading scripts in the browser.

Reviewer 2: ```suggestion
In server-side mode, tracking happens on the server, handling all events without loading scripts in the browser.
```
I'd treat "HTML placeholders and comments" as implementaiton details which are not important right now (because they are not, they are just placeholders and comments)

In client-side mode, it inserts script tags so tracking runs directly in the browser.

You can switch between server and client tracking at any time by changing the tracking type to fit your setup and needs.

## Capabilities

### Tracking

Raptor [tracking functions](tracking_functions.md) allow you to collect data about how users interact with your products and content.

You can track product visits to better understand what users are viewing.
Provided [Twig functions](../../templating/twig_function_reference/recommendations_twig_functions.md) simplify the implementation, allowing developers to quickly add tracking to templates without complex setup.

This gives you the data you need to better understand user behavior, improve recommendations, and support personalization.

### Recommendation blocks

> The Raptor Integration provides a set of ready-to-use recommendation blocks that can be added directly in the [Page Builder](page_builder_guide.md).

Reviewer 3: ```suggestion
The Raptor Integration add-on provides a set of ready-to-use recommendation blocks that can be added directly in the [Page Builder](page_builder_guide.md).
```
